### PR TITLE
SeasonTST Dataset and DataLoaders

### PR DESCRIPTION
**Pull request by paololucchino**
This PR:
- Adds packages and Makefile for linting
- Consolidates code into SeasonTST folder
- Generalizes Dataset to accept any arbitrary xr.Dataset and creates batches using xbatcher. Together with shuffle=True, this *achieves randomization of the extracted series across time and space*.
- Creates a first version of SeasonTST_pretrain.py
- Following recommendations [here](https://earthmover.io/blog/cloud-native-dataloader) uses dask cache to minimize repeat reads from zarr. 
- Other recommendations (multiprocessing and prefetch) have not been taken on board. I tried them, but results in slower load suggesting the parameters need to be optimized. Added logging to maybe help with this task.

### CODE REVIEW

File: PatchTST-env.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+name: PatchTST
+channels:
```
Comment (sem0029) [inline PatchTST-env.yml:2]: I had to add pytorch here

File: SeasonTST/utils.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
+import json
+from types import SimpleNamespace
+
+import torch
+import xarray as xr
+from torch.utils.data import Dataset
+
```
Comment (sem0029) [inline SeasonTST/utils.py:7]: I'm not sure if it's a windows thing, but I had to insert the PatchTST path explicitly for the rest of the imports to work

**Review by sem0029 — COMMENTED** [review]
Thanks @paololucchino!  I've started testing the pretrain script and so far it seems to be working, I'll update you tomorrow how it goes.

**Review by sem0029 — APPROVED** [review]
